### PR TITLE
Breakdown development workflow from PR template

### PR DESCRIPTION
**Pull request by prince-chrismc**
The goal of this PR is to reduce the burden on contributors looking to get involved.
But it's extremely important to capture the details of developing the project locally.

- Trimmed the PR template + flipped order
- new "dev workflow"
   - old template + missing steps to install rust tooling
- reworked "good pr" so be more human friendly

### CODE REVIEW

**Comment by prince-chrismc** [comment]
I asked @AbhijithGanesh for help writing a small script that will complement the leaner template.
https://github.com/pyrsia/pyrsia/issues/437

**Review by sbtaylor15 — APPROVED** [review]
Looks good.  Easier to follow with changes that were made.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
 -->
-- [ ] I've included a brief description and a good title of the proposed changes and how to test them.
 - [ ] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
-- [ ] I've associated an [issue](https://github.com/pyrsia/pyrsia/issues) with this Pull Request. 
-- [ ] I've checked that the code is up-to-date with the `pyrsia/main` branch.
 - [ ] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
-- [ ] I've assigned this Pull Request to "pyrsia/collaborators"
+- [ ] I've included a good title and brief description along with how to review them.
+- [ ] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).
+- [ ] I've requested a review  from "pyrsia/collaborators"
```
Comment (chb0github) [inline .github/pull_request_template.md:28]: If this isn't automated, it won't happen. Just saying

**Review by chb0github — DISMISSED** [review]

File: .github/pull_request_template.md (hunk 1 above)
Comment (prince-chrismc) [reply .github/pull_request_template.md:28]: It hard to automate "everything is ready for others to be assigned" there's a gap between "it's open for feedback" and "I am done working" 🤔

**Review by efrisch — APPROVED** [review]
I found a few typos and corrected them. The only thing I feel we may want to revisit is something like:
[ ] I have run the "pre-commit.sh" or "pre-commit.bat" script and it completes with no errors or warnings.

Thoughts?

Otherwise 👍 🪨 🎸

**Comment by prince-chrismc** [comment]
The "pre-commit" script is too heavy for the majority, I think only code contributors should consider it as a sanity check incase they are unsure of the development flow.

**Comment by prince-chrismc** [comment]
> I found a few typos and corrected them.

Teamwork makes the dreamwork!

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,38 +1,39 @@
...
 -->
 
-## PR Checklist
+## Description
 
-<!--
+Fixes pyrsia/pyrsia#
 
-Locally run the build process
+This PR does... by accomplishing ... and it can be review by ...
```
Comment (betarelease) [inline .github/pull_request_template.md:12]: `it can be reviewed by...`

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,38 +1,39 @@
...
-Fixes pyrsia/pyrsia#
+Locally run the build process (required for code changes, or you can remove this section).
+For a complete list of steps, check out the [developer workflow](https://github.com/pyrsia/pyrsia/blob/main/docs/dev_workflow.md)!
 
-This PR does... by accomplishing ...
+-->
+### Code Contributions
 
-## Screenshots (optional)
+- [ ] I've built the code `cargo build --workspace`.
```
Comment (betarelease) [inline .github/pull_request_template.md:38]: This should be replaced by `precommit.sh` ?

File: docs/dev_workflow.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# Code Contribution Workflow
+
+When developing new features for Pyrsia, we aim for the best quality code possible. Here's the steps with some "How To"s on getting there.
```
Comment (betarelease) [inline docs/dev_workflow.md:3]: Thanks for adding and improving this section. This was missing and as we added I was falling behind on what is the next item I should install.

**Review by betarelease — APPROVED** [review]
Added a couple of minor comments. Once you address them this can be merged.

File: .github/pull_request_template.md (hunk 2 above)
Comment (prince-chrismc) [reply .github/pull_request_template.md:12]:
```suggestion
This PR does... by accomplishing ... and it can be reviewed by ...
```

File: .github/pull_request_template.md (hunk 3 above)
Comment (prince-chrismc) [reply .github/pull_request_template.md:38]: Discussed above https://github.com/pyrsia/pyrsia/pull/430#issuecomment-1063494936

TL;DR - it too heavy for the most contributors

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -1,38 +1,39 @@
...
-Before submitting your Pull Request please check the following.
+Before submitting your Pull Request, please go over our check list.
 
 -->
 
-## PR Checklist
+## Description
 
-<!--
+Fixes pyrsia/pyrsia#
```
Comment (AbhijithGanesh) [inline .github/pull_request_template.md:10]: I think there must be a space

`Fixes pyrsia/pyrsia #`

File: .github/pull_request_template.md (hunk 3 above)
Comment (AbhijithGanesh) [reply .github/pull_request_template.md:38]: I agree with Christopher here.

**Review by AbhijithGanesh — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 5 above)
Comment (prince-chrismc) [reply .github/pull_request_template.md:10]: 🤔 no this is the full syntax from https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword



Issue in a different repository | KEYWORD OWNER/REPOSITORY#ISSUE-NUMBER | Fixes octo-org/octo-repo#100
-- | -- | --

**Review by AbhijithGanesh — APPROVED** [review]
